File: java/src/tarehart/rlbot/carpredict/CarPath.java
package tarehart.rlbot.carpredict;

import java.util.ArrayList;
import java.util.List;

public class CarPath {

    ArrayList<CarSlice> path = new ArrayList<>();

    public CarPath(CarSlice start) {
        path.add(start);
    }

    public void addSlice(CarSlice slice) {
        path.add(slice);
    }

    public List<CarSlice> getSlices() {
        return path;
    }

}

File: java/src/tarehart/rlbot/carpredict/CarPath.kt
package tarehart.rlbot.carpredict

import java.util.ArrayList

class CarPath() {

    val path = ArrayList<CarSlice>()

    val firstSlice
        get() = path[0]

    val lastSlice
        get() = path[path.size - 1]

    fun addSlice(slice: CarSlice) {
        path.add(slice)
    }

}
